code/asteroid/asteroid.cpp

```diff
@@ -1137,6 +1138,23 @@ int asteroid_check_collision(object *pasteroid, object *other_obj, vec3d *hitpos
 
 		}
 	}
+
+	// check if the hit point is beyond the clip plane if the ship is warping.
+	if (mc_ret_val) {
+		WarpEffect* warp_effect = nullptr;
+		ship* shipp = &Ships[pship_obj->instance];
+
+		// this is extremely confusing but mc.hit_point_world isn't actually in world coords
```

Comment:
Hey, if the other variable was that poorly named then we should change it now while we're thinking about it.

Reply:
Well see quite often it *is* in fact the world hit point, so it's not an inaccurate name *most of the time*. It's definitely unideal, and I looked at refactored the above collision to have it actually be a world position but it quickly became clear it would need some not insignificant changes, testing, and just got out of scope.